.gitlab-ci.yml

```
@@ -0,0 +1,148 @@
+variables:
+  DOCKER_REPOSITORY: mendersoftware/${CI_PROJECT_NAME}
+  BUILD_MASTER: "true"
+  BUILD_HOSTED: "true"
+  BUILD_2_4: "true"
+  BUILD_2_3: "true"
+
+stages:
+  - build
+  - publish
+
+include:
+  - project: 'Northern.tech/Mender/mendertesting'
+    file: '.gitlab-ci-github-status-updates.yml'
+
+build:master:
+  stage: build
+  only:
+    variables:
+      - $BUILD_MASTER == "true"
+  variables:
+    DOCKER_PUBLISH_TAG: "development"
+    DEPLOYMENTS_REPO: "deployments-enterprise"
+    DEPLOYMENTS_REV: "master"
+    DEVICEAUTH_REPO: "deviceauth"
+    DEVICEAUTH_REV: "master"
+    INVENTORY_REPO: "inventory-enterprise"
+    INVENTORY_REV: "master"
+    TENANTADM_REPO: "tenantadm"
+    TENANTADM_REV: "master"
+    USERADM_REPO: "useradm-enterprise"
+    USERADM_REV: "master"
+    WORKFLOWS_REPO: "workflows-enterprise"
+    WORKFLOWS_REV: "master"
+  tags:
+    - docker
+  image: docker
+  services:
+    - docker:19.03.12-dind
+  script:
+    - echo "building ${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_TAG}"
+    - docker build
+        --tag ${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_TAG}
+        --build-arg DEPLOYMENTS_REPO
+        --build-arg DEPLOYMENTS_REV
+        --build-arg DEVICEAUTH_REPO
+        --build-arg DEVICEAUTH_REV
+        --build-arg INVENTORY_REPO
+        --build-arg INVENTORY_REV
+        --build-arg TENANTADM_REPO
+        --build-arg TENANTADM_REV
+        --build-arg USERADM_REPO
+        --build-arg USERADM_REV
+        --build-arg WORKFLOWS_REPO
+        --build-arg WORKFLOWS_REV
+        --build-arg GITHUB_TOKEN
+        --no-cache
+        ${DOCKER_DIR:-.}
+    - docker save ${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_TAG} > ${CI_PROJECT_NAME}_${DOCKER_PUBLISH_TAG}.tar
+  artifacts:
+    expire_in: 2w
+    paths:
+      - ${CI_PROJECT_NAME}_${DOCKER_PUBLISH_TAG}.tar
+
+build:hosted:
+  extends: build:master
+  only:
+    variables:
+      - $BUILD_HOSTED == "true"
+  variables:
+    DOCKER_PUBLISH_TAG: "hosted"
+    DEPLOYMENTS_REV: "staging"
+    DEVICEAUTH_REV: "staging"
+    INVENTORY_REV: "staging"
+    TENANTADM_REV: "staging"
+    USERADM_REV: "staging"
+    WORKFLOWS_REV: "staging"
+
+build:2.4:
+  extends: build:master
+  only:
+    variables:
+      - $BUILD_2_4 == "true"
+  variables:
+    DOCKER_PUBLISH_TAG: "2.4"
+    DEPLOYMENTS_REV: "2.0.x"
+    DEVICEAUTH_REV: "2.3.x"
+    INVENTORY_REPO: "inventory"
+    INVENTORY_REV: "2.0.x"
+    TENANTADM_REV: "2.0.x"
+    USERADM_REV: "1.11.x"
+    WORKFLOWS_REV: "1.1.x"
+
+build:2.3:
+  extends: build:master
+  only:
+    variables:
+      - $BUILD_2_3 == "true"
+  variables:
+    DOCKER_PUBLISH_TAG: "2.3"
+    DEPLOYMENTS_REV: "1.9.x"
+    DEVICEAUTH_REV: "2.2.x"
+    INVENTORY_REPO: "inventory"
+    INVENTORY_REV: "1.7.x"
+    TENANTADM_REV: "1.1.x"
+    USERADM_REV: "1.10.x"
+    WORKFLOWS_REPO: "workflows"
+    WORKFLOWS_REV: "1.0.x"
+
+publish:master:
+  stage: publish
+  only:
+    refs:
+      - master
```

If you look close to the acceptance criteria in [MEN-3296](https://tracker.mender.io/browse/MEN-3296), we planned to have two branches `master` and `production` (similar as we do for mender-docs-site). So here we need to add
```suggestion
publish:master:
  stage: publish
  only:
    refs:
      - master
      - production
```

---

right, done